### ISSUE TITLE
Move bootstrap to peerDependencies

### ISSUE THREAD

**Issue by xvaara** [cdmoro/bootstrap-vue-3#236]
I think bootstrap should be in peerDependencies, so that bootstrap-vue-3 doesn't install it's own conflicting one in node_modules/bootstrap-vue-3/node_modules/bootstrap.

The instructions already say `yarn add bootstrap bootstrap-vue-3`

**Comment by Radouch** [comment]
I am not sure if I understand this topic correctly, but: I am trying to develop our own custom library with just few special components, based on Vue 3 + Bootstrap 5. As this library should be used in Vue 3 + Bootstrap 5 applications, any of its component does not import Bootstrap nor Bootstrap-Icons (I use Storybook for development, so Bootstrap is imported in stories, not in component itself, mimicking the real application inside which my component will be used. So the build size of my library is relatively small (~ tens of kBs).

Now I tried to incorporate `bootstrap-vue-3` library to use its `BTable` component. Even if I import only BTable component with `import { BTable } from "bootstrap-vue-3";` build size of my library increased to 1+ MB as `BTable` "injected" whole Bootstrap and especially Bootstrap Icons into the build.

It would be great if Bootstrap could be really externalized from `bootstrap-vue-3`.

**Comment by mariusa** [comment]
Build size is fixed now.

There is no `node_modules/bootstrap-vue-3/node_modules/bootstrap`
Since this lib can't be used without bootstrap, it makes sense for this lib to pull it too. Instructions are wrong, I'll update that to install only `bootstrap-vue-3`

**Comment by Daedalus11069** [comment]
@mariusa I was following this issue since I agreed(although for different reasons, which I'll get to) with this proposal.

While it makes sense that the library should pull it, it doesn't make sense for the library to include bootstrap in the built files; for people already including bootstrap in their build process, it adds unnecessary size and conflicts to the library.

I of course can't speak for everyone, and only my current pipeline, but for instance: A site I'm working on supports extensions to the core; the site in question already includes bootstrap's css and js.  If I include bootstrap-vue-3 in my extension, the two versions of bootstrap's js(one in the site, and the one in bootstrap-vue-3) conflict, preventing a collapse from working.

Take [this fiddle](https://jsfiddle.net/Daedalus/7bzLmv1j/) for example; the code to reproduce it is simple and will be included in  this post, but clicking the collapse opens it as normal.  Clicking the button again to close it fails to work, however.

I did test this before writing this post and bootstraps' js is still in the built files.

<details>
  <summary>Code to reproduce; make sure bootstrap.js is loaded into the document.</summary>

```html
<div id="app">
  <b-button v-b-toggle.collapse-1 variant="primary">Toggle Collapse</b-button>
  <b-collapse id="collapse-1" class="mt-2">
    <b-card>
      <p class="card-text">Collapse contents Here</p>
      <b-button v-b-toggle.collapse-1-inner size="sm">Toggle Inner Collapse</b-button>
      <b-collapse id="collapse-1-inner" class="mt-2">
        <b-card>Hello!</b-card>
      </b-collapse>
    </b-card>
  </b-collapse>
</div>
```

```js
window.addEventListener('DOMContentLoaded', () => {
  const app = Vue.createApp({});
  app.use(window['bootstrap-vue-3'].BootstrapVue3);
  app.mount(document.querySelector("#app"));
});
```
</details>

If bootstrap is already in the page, conflicts will happen; if it is removed for the build process, this issue is fixed.  As it stands, I've had to clone the library and build it myself so I could use it in my projects because of these conflicts, but that of course is a sub-optimal solution.

<details>
  <summary>The simple change I made to the process to avoid conflicts in vite.config.ts</summary>

```js
//...
external: ['vue', 'bootstrap'],
//...
output: {
//...
  globals: {
    vue: 'Vue',
    bootstrap: 'bootstrap',
  },
},
//...
```
</details>

TL;DR: If bootstrap is already included in the page, the library breaks.  Please move it to peerDependencies and exclude it from the build process.

**Comment by mariusa** [comment]
Got it

**Comment by mariusa** [comment]
Guys, we're getting a build error which seems to be due to peerDependencies:
https://github.com/cdmoro/bootstrap-vue-3/pull/277#issuecomment-1038463985

Any suggestions on how to fix that? I can't figure it out. Thanks!

**Comment by mariusa** [comment]
We can surely require bootstrap 5.1.x . Question is, what's the best solution while keeping bootstrap in peerDependencies?
